File: transport/server/play-server/src/main/scala/play/core/server/ServerStartException.scala
/*
 * Copyright (C) from 2022 The Play Framework Contributors <https://github.com/playframework>, 2011-2021 Lightbend Inc. <https://www.lightbend.com>
 */

package play.core.server

/**
 * Indicates an issue with starting a server, e.g. a problem reading its
 * configuration.
 */
final case class ServerStartException(message: String, cause: Option[Throwable] = None)
    extends Exception(message, cause.orNull)

File: dev-mode/play-run-support/src/main/java/play/runsupport/ServerStartException.java
/*
 * Copyright (C) from 2022 The Play Framework Contributors <https://github.com/playframework>, 2011-2021 Lightbend Inc. <https://www.lightbend.com>
 */

package play.runsupport;

public class ServerStartException extends IllegalStateException {

  public ServerStartException(Throwable underlying) {
    super(underlying);
  }

  @Override
  public String getMessage() {
    return getCause().getMessage();
  }
}
